modules/internal/ChapelUtil.chpl

```diff
@@ -180,4 +180,10 @@ module ChapelUtil {
 
     chpl_moduleDeinitFuns = nil;
   }
+
+  // TODO: Fix 'new Private()' leak -- Discussed in #6726
+  use PrivateDist;
+  proc deinit() {
```

Comment:
What prevents this deinit from being in privatedist?

Reply:
valgrind errors.

If you do that the following events occur in PrivateDist module when its first imported
- init `chpl_privateDist` (an unmanaged class)
- init `PrivateSpace` (a record)

when the program ends
- deinit PrivateDist module
  - deinit chpl_privateDist
- deinit `PrivateSpace`
  - deinit chpl_privateDist (whoops, double free)

See https://github.com/chapel-lang/chapel/issues/6726